example/src/shared/SubSideNav.js

```diff
@@ -0,0 +1,56 @@
+import React from "react";
+import {
+  SubNav,
+  SubNavHeader,
+  SubNavSections,
+  SubNavSection,
+  SubNavLink,
+} from "@strapi/design-system/SubNav";
+import { Box } from "@strapi/design-system/Box";
+import { TextButton } from "@strapi/design-system/TextButton";
+import AddIcon from "@strapi/icons/AddIcon";
+import { cmData } from "./cmData";
+import { useSearch } from "./useSearch";
+
+export const SubSideNav = () => {
+  const { search, setSearch, searchResults } = useSearch(cmData, { keys: ["label"] });
+  const collectionType = searchResults.filter((link) => link.kind === "collectionType");
+  const singleType = searchResults.filter((link) => link.kind === "singleType");
+
+  return (
+    <SubNav ariaLabel="Builder sub nav">
+      <SubNavHeader
+        searchable
+        value={search}
+        onClear={() => setSearch("")}
+        onChange={(e) => setSearch(e.target.value)}
+        label="Builder"
+        searchLabel="Search..."
+      />
+      <SubNavSections>
+        <SubNavSection label="Collection Type" collapsable badgeLabel={collectionType.length.toString()}>
+          {collectionType &&
+            collectionType.map((link) => (
+              <SubNavLink href={link.to} active={link.active} key={link.id}>
```

Comment:
Can you try to make it work with a NavLink from react-router-dom? Or do you prefer to wait for the PR that switches from reach-router to react-router-dom?